### PR TITLE
Implements a Rake task for importing vocabularies from a CSV file

### LINKED ISSUE #233: Rake task to import vocabs from CSV

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #233

### CODE REVIEW

**Comment by jrgriffiniii** [comment]
This is now prepared for an initial review.

File: app/queries/query_adapter/ephemera_vocabulary_query_adapter.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+module QueryAdapter
+  class EphemeraVocabularyQueryAdapter < Base
+    def find_by(label:, vocabulary: nil)
```
Comment (tpendragon) [inline app/queries/query_adapter/ephemera_vocabulary_query_adapter.rb]: I think a generic `find_by` property which has multiple values you can pass it is setting a dangerous precedent. A special class for every model type seems like an early abstraction too which could lead to a lot of confusion about where to put some code and what interface is being expected at each layer. What's the import code look like without the class-specific `QueryAdapter`s? Is it also complex?

File: app/services/persistence_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    change_set.validate(resource.attributes)
+    yield change_set if block_given?
+    # change_set.member_of_vocabulary_id = vocabulary.id if vocabulary.present?
+    return false unless change_set.sync
+    @change_set_persister.save(change_set: change_set)
+  end
+
+  private
+
+    def change_set_class_name
```
Comment (tpendragon) [inline app/services/persistence_adapter.rb]: You can just make this always `DynamicChangeSet`

File: app/services/persistence_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    @change_set = change_set
+  end
+
+  def create(*args)
+    resource = @model.new(*args)
+    change_set = change_set_class.new(resource)
+    change_set.validate(resource.attributes)
+    yield change_set if block_given?
+    # change_set.member_of_vocabulary_id = vocabulary.id if vocabulary.present?
+    return false unless change_set.sync
```
Comment (tpendragon) [inline app/services/persistence_adapter.rb]: I don't think this ever returns false - validate above does.

File: app/services/persistence_adapter.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    @model = model
+    @change_set = change_set
+  end
+
+  def create(*args)
+    resource = @model.new(*args)
+    change_set = change_set_class.new(resource)
+    change_set.validate(resource.attributes)
+    yield change_set if block_given?
+    # change_set.member_of_vocabulary_id = vocabulary.id if vocabulary.present?
```
Comment (tpendragon) [inline app/services/persistence_adapter.rb]: Delete?

File: app/services/ingest_vocab_service.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    @term_queries ||= QueryAdapter.new(query_service: query_service, model: EphemeraTerm)
+  end
+
+  def term_persistence
+    @term_persistence ||= PersistenceAdapter.new(change_set_persister: change_set_persister, model: EphemeraTerm)
+  end
+
+  def ingest_vocabulary(label:, parent_vocab: nil)
+    return unless label
+    category = vocab_queries.find_with(FindEphemeraVocabularyByLabel, label: label).first
```
Comment (tpendragon) [inline app/services/ingest_vocab_service.rb]: Soon this'll be able to become `vocab_queries.custom_queries.find_ephemera_vocabulary_by_label(label: label)`, so I'm not 100% sure the `QueryAdapter` abstraction is necessary.

**Comment by tpendragon** [comment]
Looking cleaner! Just a few more notes.

**Pull request merged**